inbm/dispatcher-agent/dispatcher/sota/os_factory.py

```diff
@@ -137,10 +139,12 @@ def create_os_updater(self) -> OsUpdater:
         logger.debug("")
         return YoctoX86_64Updater()
 
-    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], proceed_without_rollback: bool) -> Snapshot:
+    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], 
+                           proceed_without_rollback: bool, device_reboot: str) -> Snapshot:
         logger.debug("")
         trtl = Trtl(PseudoShellRunner(), BTRFS)
-        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, proceed_without_rollback)
+        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, 
+                             proceed_without_rollback, device_reboot)
```

Comment:
**Suggestion:** Consider handling the case where 'device_reboot' is set to 'no' but a reboot is still triggered due to a failure in the update process. This can be done by adding a conditional check before initiating the reboot. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], 
                           proceed_without_rollback: bool, device_reboot: str) -> Snapshot:
        logger.debug("Creating snapshotter with device_reboot set to {}".format(device_reboot))
        trtl = Trtl(PseudoShellRunner(), BTRFS)
        snapshotter = YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, 
                                    proceed_without_rollback, device_reboot)
        if device_reboot.lower() == 'no':
            snapshotter.disable_reboot_on_failure()
        return snapshotter
```